tests/providers/microsoft/azure/hooks/test_azure_data_factory.py

```diff
@@ -177,8 +177,9 @@ def test_create_factory(hook: AzureDataFactoryHook, user_args, sdk_args):
     implicit_factory=((MODEL,), (DEFAULT_RESOURCE_GROUP, DEFAULT_FACTORY, MODEL)),
 )
 def test_update_factory(hook: AzureDataFactoryHook, user_args, sdk_args):
-    hook._factory_exists = Mock(return_value=True)
-    hook.update_factory(*user_args)
+    with patch.object(hook, "_factory_exists") as mock_factory_exists:
```

Comment:
Hmmm. I really like this change. Somehow using generic `Mock` object for this case felt a bit weird. Seems like MyPy also forces on us a bit nicer approach for mocking.

Reply:
@uranusjr  - WDYT?